### DISCUSSION TITLE
Discussion on moving from react to yew

### DISCUSSION THREAD

**Discussion by reibomaru** [team-azb/route-bucket-frontend#105] [Ideas]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by reibomaru** [comment]
## pagesごとのコンポーネントの依存関係[@master](https://github.com/team-azb/route-bucket-frontend)
![graph](https://user-images.githubusercontent.com/46376570/168241088-f9adcef0-6b3b-48d1-9e4a-c5eff232e500.svg)

**Reply by reibomaru** [reply]
Topとそれ以外でしか分けられなさそう。。

**Reply by reibomaru** [reply]
`contexts/AuthenticationProvider/index.tsx`と`organisms/SignInRequiredTemplate/index.tsx`を依存関係から外してみた場合。
![graph](https://user-images.githubusercontent.com/46376570/168242712-d70de057-4308-40ae-bb27-414c267d6598.svg)

**Reply by reibomaru** [reply]
4つに分けれた。

**Reply by reibomaru** [reply]
そこからさらにatomsを依存関係から外してみた場合
![graph](https://user-images.githubusercontent.com/46376570/168243445-11d27d44-8c6e-4a68-9618-add5b41cc6c7.svg)

**Reply by reibomaru** [reply]
`molecules/InputWithError/index.tsx`を外してみる(moleculesにあたるのはこれ一つだけ)
![graph](https://user-images.githubusercontent.com/46376570/168244417-e39d31b2-8931-493f-9346-a65e1a9aba10.svg)

**Reply by reibomaru** [reply]
いいかんじ、ただし、これらはすべてtsファイルを依存関係からはずして考えているのでtsファイルについて考える。

**Reply by shugo256** [reply]
うおおおすげえなんだこれ

**Reply by reibomaru** [reply]
最初手作業で調べようと思ったけど、なかなか限界あったから、調べてみたところ
https://github.com/pahen/madge という神ライブラリをみつけた。
これつかうとJavaScriptの依存関係全部洗ってくれている。いい感じの画像もつくってくれる。

**Reply by reibomaru** [reply]
tsファイルの依存関係は以下
![graph](https://user-images.githubusercontent.com/46376570/168249889-de49d9fe-ee96-4136-bab2-97b5918de584.svg)

**Reply by reibomaru** [reply]
`api/helpers.ts`と`types.ts`を片付ければこんな感じ
![graph](https://user-images.githubusercontent.com/46376570/168250483-9b3a6145-27a6-481c-a9fb-a86078c21a38.svg)

**Comment by reibomaru** [comment]
## yew移行への手順（暫定）
1. tsファイルの移行
    1. `api/helpers.ts`と`types.ts`を書き換える
    2. 以下のまとまりごとにファイルを片付ける
    ![image](https://user-images.githubusercontent.com/46376570/168252714-80ca41e3-4a66-444f-9f0c-32901990f303.png)
2. tsxファイルとcssの移行
tsxファイルと必要であればそれに関係するcssファイルを移行する
    1. `atoms/`を書き換える
    2. `molecules/`を書き換える
    4. `contexts/AuthenticationProvider/index.tsx`と`organisms/SignInRequiredTemplate/index.tsx`を書き換える
    5.  以下のまとまりごとに書き換える
    ![image](https://user-images.githubusercontent.com/46376570/168252921-bf60d89d-26ce-423f-aab8-585922f56939.png)
